pkg/streamingpromql/engine.go

```diff
@@ -44,12 +47,26 @@ func NewEngine(opts EngineOpts, limitsProvider QueryLimitsProvider, metrics *sta
 		return nil, errors.New("enabling delayed name removal not supported by Mimir query engine")
 	}
 
+	// Sort DisabledFunctions to optimise lookups
```

Comment:
I would make this comment (and the one below) even clearer: we must sort these two slices because we use a binary search over them later on.